meta-balena-common/classes/kernel-resin.bbclass

```diff
@@ -875,7 +872,7 @@ python do_kernel_resin_checkconfig() {
 
             for config in notconfigured:
                 if not config.endswith('=n'):
-                    bb.warn("Checking for %s in the kernel configs failed for %s."
+                    bb.fatal("Checking for %s in the kernel configs failed for %s."
```

Comment:
I see this one warn all the time for certain devices, so it needs to be a warn only unless we go and clean up all the old kernel configs that are no longer valid.

Reply:
Yes, device types that don't build need to be fixed.

Reply:
@alexgg @klutchell Maybe we need `required` and `recommended` config groups? That would allow us to require certain configs like overlay without breaking existing builds that treat this like a warning.

Reply:
@jakogut more than recommended we need a way to enforce depending on kernel version. The purpose of this class should be to make sure that the kernel contains all the configuration the product needs. The problem is that different kernel versions will need different configs.